### DISCUSSION TITLE
Cannot embed superset in ReactJS app

### DISCUSSION THREAD

**Discussion by kornelc** [apache/superset#20267] [Q&A / Help]
Hello,

I have been trying to wrap my brain around embedding Superset in a ReactJS app. All pointers seem to lead me to Embedded SDK, which has the following documentation:

embedDashboard({
  id: "abc123", // given by the Superset embedding UI
  supersetDomain: "https://superset.example.com",
  mountPoint: document.getElementById("my-superset-container"), // any html element that can contain an iframe
  fetchGuestToken: () => fetchGuestTokenFromBackend(),
  dashboardUiConfig: { hideTitle: true }, // dashboard UI config: hideTitle, hideTab, hideChartControls (optional)
});

My questions:
- What is "the Superset <b><I>embedding</I></b> UI"? Is this the Superset web app that we use to create the dashboards? I.e. is this the normal superset app (the app in the superset docker image), or there is some other "embedding UI" app? Are we supposed to copy the ID from the URL of a dashboard? Those IDs are numeric and they don't look like abc123. They look like 123.  
- supersetDomain: Is this where the superset web app is running, so for example the host where the docker container is running? Can this be a different domain/host? Won't we run into <b><I>cross site scripting issues</I></b>, or the embedding SDK takes care of that? (When using an iframe directly I ran into numerous cross site scripting issues.)
- mountPoint: "my-superset-container" is this the ID of an HTML tag? If yes, how does that work in a ReactJS app (where I would want to show a chart inside a React widget), or any other one page app, since there <b><I>we don't normally access the tags directly by id</I></b>? React manages the tags for us. The entire react app runs in one HTML tag with the id "root". So what do I use as a container.
- fetchGuestToken: an <b><I>example</I></b> of this function would be great, one that shows what this call looks like

I have been looking for a full example of embedding Superset in an app, where the host app does the authentication (so no login prompt from Superset), either with or without using Embedding SDK but I haven't been able to find anything that has enough info to get started.

**Comment by cwegener** [comment]
> What is "the Superset embedding UI"?

When you are logged in to Superset and open a dashboard, the dashboard settings menu (three dots) will have an entry in it called "Embed Dashboard" https://github.com/apache/superset/pull/19364

> supersetDomain: Is this where the superset web app is running

It's the URL where your Superset instance is published at.

> Won't we run into cross site scripting issues,

That depends on your application, I guess. Don't use unsafe Javascript function calls. React will protect you from doing that (mostly).

> mountPoint: "my-superset-container" is this the ID of an HTML tag

Any element that's available in the DOM will work. An empty `<div>` is the most common type of element to use for that. Yes, you will need address the element. `document.getElementById()` would be one way to reference the element.

Without being able to create such a container element in your app, you will have difficulties setting up the embedded SDK.

> fetchGuestToken: an example of this function

https://superset.apache.org/docs/api
https://github.com/apache/superset/tree/master/superset-embedded-sdk#creating-a-guest-token


The design spec is also an important document to read.
https://github.com/apache/superset/issues/17187
